flow/layers/physical_model_layer.cc

```diff
@@ -0,0 +1,53 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/flow/layers/physical_model_layer.h"
+
+#include "third_party/skia/include/utils/SkShadowUtils.h"
+
+namespace flow {
+
+PhysicalModelLayer::PhysicalModelLayer()
+    : rrect_(SkRRect::MakeEmpty()) {}
+
+PhysicalModelLayer::~PhysicalModelLayer() {}
+
+void PhysicalModelLayer::Preroll(PrerollContext* context, const SkMatrix& matrix) {
+  PrerollChildren(context, matrix);
+  if (!context->child_paint_bounds.intersect(rrect_.getBounds()))
+    context->child_paint_bounds.setEmpty();
+
+  // Add some margin to the paint bounds to leave space for the shadow.
+  SkRect bounds(context->child_paint_bounds);
+  bounds.outset(50.0, 50.0);
```

Comment:
This is a hardcoded estimate of the upper bounds for where DrawShadow will paint.  AFAIK Skia doesn't have a way to calculate this precisely.  Added a comment.